### ISSUE TITLE
Touchpad Support

### ISSUE THREAD

**Issue by thestr4ng3r** [thestr4ng3r/chiaki#270]
High-level status of touchpad support.

- [x] Implement the Protocol
- GUI
    - [ ] Virtual Touchpad
    - [x] Physical Controller Touchpad on Linux
    - [ ] Physical Controller Touchpad on Windows
    - [ ] Physical Controller Touchpad on macOS
- Android
    - [ ] Virtual Touchpad
    - [ ] Physical Controller Touchpad

**Comment by zeparu** [comment]
whats the current status with this? i read in the patchnotes that support for physical controllers on linux was added, i have the `chiaki-git` package from the AUR, so its built from source too, but it seems pressing down the touchpad still doesnt work properly and is not being sent to the ps4

**Comment by thestr4ng3r** [comment]
Does only the touchpad button not work or the entire touchpad input?

A little background about how it can be used currently: The Linux kernel comes with a nice builtin driver for the DualShock 4's touchpad which can be accessed through evdev. Chiaki now has a built-in library which provides a nice interface around this: https://github.com/thestr4ng3r/chiaki/tree/master/setsu It will be automatically enabled if the right dependencies are found (libevdev and libudev). To force-enable it you can pass `-DCHIAKI_ENABLE_SETSU=ON` to cmake.

tl;dr: Install libevdev and build with `cmake -DCHIAKI_ENABLE_SETSU=ON (...)`.

**Comment by zeparu** [comment]
i built it explicitly with the parameter now, but at least the touchpad press is not sent properly

my first assumption would be, that gnome is "stealing" the touchpad input first, to use as a touchpad mouse input
i can disable the touchpad in the gnome settings, but then, as far as `xev` shows, the touchpad is completely disabled, probably in evdev directly, so depending on the setting in gnome it either will work as mouse in gnome, or will be disabled

**Comment by thestr4ng3r** [comment]
You can see the raw evdev input by doing `evtest /dev/input/by-id/usb-Sony_Computer_Entertainment_Wireless_Controller-event-mouse` (make sure to use the `-event-mouse` device to get the touchpad)

To disable the touchpad acting as a mouse, for X you can try https://wiki.archlinux.org/index.php/Gamepad#Disable_touchpad_acting_as_mouse
for wayland, it depends on the compositor. In Sway, you can put this in your config for example:
```
# Gen 1 DualShock 4
input "1356:1476:Sony_Computer_Entertainment_Wireless_Controller_Touchpad" {
	events disabled
}

# Gen 2 DualShock 4
input "1356:2508:Sony_Interactive_Entertainment_Wireless_Controller_Touchpad" {
	events disabled
}

# Gen 2 DualShock 4 (Bluetooth)
input "1356:2508:Wireless_Controller_Touchpad" {
	events disabled
}
```
I don't know about Gnome, but since you use `xev` you may be running X anyway.

**Comment by zeparu** [comment]
i manually disabled it, as described in the arch wiki, and its not working as a mouse anymore, but i chiaki is still not properly sending it to the ps4 (mouse click on window or selected keybind, default T still work)

**Comment by Fredrum** [comment]
It is working on Raspbian Buster with my RPi4.

**Comment by wpinacz** [comment]
Hi all,
I've got problem touchpad working too, on Ubuntu 20.04, touchpad not working, only on "mouse mode" click works (mouse click works too so it's emulating only).
The only difference is that evtest works but on different device, I don't have device:
/dev/input/by-id/usb-Sony_Computer_Entertainment_Wireless_Controller-event-mouse
I have:
/dev/input/by-id/usb-Sony_**Interactive**_Entertainment_Wireless_Controller-**if03**-event-mouse
evtest works with above device.
Maybe it's the difference problem?
